src/Private/Responses.ps1

```diff
@@ -48,4 +82,403 @@ function Show-PodeErrorPage {
+        [string]
+        $ContentType = $null,
+
+        [Parameter()]
+        [int]
+        $MaxAge = 3600,
+
+        [Parameter()]
+        [int]
+        $StatusCode = 200,
+
+        [switch]
+        $Cache,
+
+        [switch]
+        $FileBrowser
+    )
+
+    # Attempt to retrieve information about the path
+    $pathInfo = Test-PodePath -Path $Path -Force -ReturnItem -FailOnDirectory:(!$FileBrowser)
+
+    if (!$pathinfo) {
+        return
+    }
+
+    # Check if the path is a directory
+    if ( $pathInfo.PSIsContainer) {
+        # If directory browsing is enabled, use the directory response function
+        Write-PodeDirectoryResponseInternal -Path $Path
+    }
+    else {
+        # are we dealing with a dynamic file for the view engine? (ignore html)
+        # Determine if the file is dynamic and should be processed by the view engine
+        $mainExt = Get-PodeFileExtension -Path $Path -TrimPeriod
+
+        # generate dynamic content
+        if (![string]::IsNullOrWhiteSpace($mainExt) -and (
+        ($mainExt -ieq 'pode') -or
+        ($mainExt -ieq $PodeContext.Server.ViewEngine.Extension -and $PodeContext.Server.ViewEngine.IsDynamic)
+            )
+        ) {
+            # Process dynamic content with the view engine
+            $content = Get-PodeFileContentUsingViewEngine -Path $Path -Data $Data
+
+            # Determine the correct content type for the response
+            # get the sub-file extension, if empty, use original
+            $subExt = [System.IO.Path]::GetExtension($pathInfo.BaseName).TrimStart('.')
+
+            $subExt = (Protect-PodeValue -Value $subExt -Default $mainExt)
+
+            $ContentType = (Protect-PodeValue -Value $ContentType -Default (Get-PodeContentType -Extension $subExt))
+
+            # Write the processed content as the HTTP response
+            Write-PodeTextResponse -Value $content -ContentType $ContentType -StatusCode $StatusCode
+        }
+        # this is a static file
+        else {
+            try {
+                if (Test-PodeIsPSCore) {
+                    $content = (Get-Content -Path $Path -Raw -AsByteStream)
+                }
+                else {
+                    $content = (Get-Content -Path $Path -Raw -Encoding byte)
+                }
+                # Determine and set the content type for static files
+                $ContentType = Protect-PodeValue -Value $ContentType -Default (Get-PodeContentType -Extension $mainExt)
+                # Write the file content as the HTTP response
+                Write-PodeTextResponse -Bytes $content -ContentType $ContentType -MaxAge $MaxAge -StatusCode $StatusCode -Cache:$Cache
+                return
+            }
+            catch [System.UnauthorizedAccessException] {
+                $statusCode = 401
+            }
+            catch {
+                $statusCode = 400
+            }
+            # If the file does not exist, set the HTTP response status to 404 Not Found
```

Comment:
Last part of the comment needs changing, to reference something like "set the HTTP response status code appropriately"